src/Illuminate/Container/Container.php

```diff
@@ -710,7 +710,7 @@ public function build($concrete)
         // hand back the results of the functions, which allows functions to be
         // used as resolvers for more fine-tuned resolution of these objects.
         if ($concrete instanceof Closure) {
-            return $concrete($this, last($this->with) ?: []);
+            return $concrete($this, last($this->with));
```

Comment:
Using `last` here means that the composer.json needs to be updated to include `illuminate/support`